### ISSUE TITLE
U-net Image Segmentation Multi-class

### ISSUE THREAD

**Issue by vonarchimboldi** [keras-team/keras#10836]
I'm trying to do an image segmentation problem where I want to segment 5 objects in an image. I'm using a U-net architecture. My final layer looks like this:

conv_final = Conv2D(OUTPUT_MASK_CHANNELS, (1, 1))(up_conv_224)
conv_final = Activation('sigmoid')(conv_final)

model = Model(inputs, conv_final, name="ZF_UNET_224")
However I get an error saying:

ValueError: Error when checking target: expected conv2d_24 to have shape (224, 224, 5) but got array with shape (224, 224, 3)

This is the generator that I'm using

    image_generator = train_datagen.flow_from_directory(
        'data/train',  # this is the target directory
        target_size=(224, 224),
        color_mode = 'rgb',# all images will be resized to 150x150
        batch_size=batch_size,
        class_mode=None,
        seed = 1)  # since we use binary_crossentropy loss, we need binary labels

# this is a similar generator, for validation data
    mask_generator = mask_datagen.flow_from_directory(
        'data/train',
        target_size=(224, 224),
        color_mode = 'rgb',
        batch_size=batch_size,
        class_mode=None,
        seed = 1)

    train_generator = zip(image_generator, mask_generator)
What can I do to fix this? Any help appreciated!

**Comment by greed2411** [comment]
If I'm not wrong you are passing images which only have three channels, `color_mode='rgb'` so therefore keras expects `(224, 224, 3)` where the three represents RGB channels, somewhere you are making the model expect `(224, 224, 5)` which means, it's looking for five channels when actually there are only three channels in reality.
If you post the entire code I'd be happy to help.

**Comment by vonarchimboldi** [comment]
Hello, thanks for your response. I was able to figure out that the Keras ImageDataGenerator and flow_from_directory do not support images with 5 channels, so I had to write a custom generator for that purpose.

I have a different problem now though. I took this network with pretrained weights that was trained on the same set of images but the end goal was different. The network was trained to produce a single output, while I need to learn a mask with all 5 objects I need segmented in the image. I thought simply popping the final layer of this network with a Conv2D(num_classes, (1,1), activation='sigmoid) would do the trick but I'm getting really bad performance. I have only 24 images to train on and one to test on so I thought transfer learning would be a great idea. Any ideas about how to debug this problem? I'd be happy to post some results from the images and masks if you'd like to help.

The pretrained network is available here: https://github.com/ZFTurbo/ZF_UNET_224_Pretrained_Model/blob/master/zf_unet_224_model.py

I just changed the output layer.
